cerbero/build/fridge.py

```diff
@@ -0,0 +1,123 @@
+# cerbero - a multi-platform build system for Open Source software
+# Copyright (C) 2012 Andoni Morales Alastruey <ylatuya@gmail.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Library General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+import os
+import traceback
+import tarfile
+
+from cerbero.errors import PackageNotFoundError, BuildStepError, FatalError
+from cerbero.utils import _, shell
+from cerbero.build.recipe import Recipe, BuildSteps
+from cerbero.utils import messages as m
+from cerbero.utils.shell import upload_curl, download_curl
+from cerbero.packages.disttarball import DistTarball
+from cerbero.packages import PackageType
+
+class Fridge (object):
```

Comment:
Ok, already commented but to clarify: If we use a Mixin, it means that a recipe can overwrite the binary methods, it makes sense for the classic build methods but not for the package/upload/download/extract steps, those must be immutable